packages/react-magma-dom/src/components/Dropdown/DropdownExpandableMenuButton.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import styled from '../../theme/styled';
+import { AccordionButton, AccordionButtonProps } from '../Accordion';
+import { IconWrapper, menuBackground } from './DropdownMenuItem';
+import { IconProps } from 'react-magma-icons';
+import { ThemeContext } from '../../theme/ThemeContext';
+import { DropdownContext } from './Dropdown';
+import { DropdownExpandableContext } from './DropdownExpandableMenuGroup';
+import { useForkedRef } from '../../utils';
+
+export interface DropdownExpandableMenuButtonProps
+  extends AccordionButtonProps {
+  icon?: React.ReactElement<IconProps>;
+  testId?: string;
+}
+
+const StyledAccordionButton = styled(AccordionButton)<{
+  hasIcon?: boolean;
+  icon?: React.ReactElement<IconProps>;
+}>`
+  font-weight: 400;
+  padding: ${props =>
+    !props.icon && props.hasIcon
+      ? `${props.theme.spaceScale.spacing03} ${props.theme.spaceScale.spacing05} ${props.theme.spaceScale.spacing03} ${props.theme.spaceScale.spacing11}`
+      : `${props.theme.spaceScale.spacing03} ${props.theme.spaceScale.spacing05}`};
+  margin: 0;
+  border-top: 0;
+  &:hover,
+  &:focus {
+    background: ${menuBackground};
+  }
+  > span {
+    display: flex;
+  }
+`;
+
+const StyledIconWrapper = styled(IconWrapper)`
+  justify-content: center;
+  /* align-items: center; */
+`;
+
+export const DropdownExpandableMenuButton = React.forwardRef<
+  HTMLDivElement,
+  DropdownExpandableMenuButtonProps
+>((props, forwardedRef) => {
+  const { children, icon, testId, ...other } = props;
+
+  const theme = React.useContext(ThemeContext);
+  const context = React.useContext(DropdownContext);
+  const expandableContext = React.useContext(DropdownExpandableContext);
+
+  const ownRef = React.useRef<HTMLDivElement>();
+  const ref = useForkedRef(forwardedRef, ownRef);
+
+  React.useEffect(() => {
+    context.registerDropdownMenuItem(context.itemRefArray, ownRef);
+  }, []);
+
+  return (
+    <StyledAccordionButton
+      {...other}
+      ref={ref}
+      icon={props.icon}
+      theme={theme}
+      hasIcon={expandableContext.hasIcon}
+      isInverse={context.isInverse}
+      testId={testId}
+    >
+      {icon && (
+        <StyledIconWrapper theme={theme} isInverse={context.isInverse}>
+          {icon}
+        </StyledIconWrapper>
+      )}
+      {children}
```

Comment:
Do we need to do something like this to add extra space for cases when the icon is missing?
```suggestion
      {!icon && menuGroupContext.hasIcons && (
         <Spacer size={40} axis={SpacerAxis.horizontal} />
      )}
      {children}
```

Reply:
There's already styling to account for a combination of an icon item and a text only item in the same group. Is that what you mean?